packages/ERTP/src/paymentLedger.js

```diff
@@ -291,7 +291,22 @@ export const makePaymentLedger = (
    */
   const withdraw = (currentBalance, updatePurseBalance, amount) => {
     amount = coerce(amount);
-    const newPurseBalance = subtract(currentBalance, amount);
+    let newPurseBalance;
+    try {
+      newPurseBalance = subtract(currentBalance, amount);
+    } catch (err) {
+      if (err instanceof RangeError) {
+        const withdrawalError = Error(
+          // @ts-ignore `X` (aka `details`) lazily constructs a
+          // string, but the types do not reflect that
+          X`Withdrawal of ${amount} failed because the purse only contained ${currentBalance}`,
+        );
+        assert.note(withdrawalError, X`Caused by: ${err}`);
```

Comment:
```suggestion
        assert.note(withdrawalError, X`Caused by: ${q(err)}`);
```

I am not sure why this isn’t a TypeScript error, since `err` is neither `string` nor `Details`.

Reply:
Doesn't `${}` coerce to a string? For what it is worth, this is cargo-culted from https://github.com/Agoric/agoric-sdk/blob/42bd6e8add36cc60515e92a4e5c3656fb6aad226/packages/eventual-send/src/track-turns.js#L52-L52

Reply:
Yes, it does coerce to string.

I recall `tsc` looking down its nose on this, but I can't seem to find those details again.

Reply:
We use 
```js
`${expr}`
``` to coerce arbitrary things to strings a lot. If TS started complaining about this, it would cause pervasive problems. Good that it does not.